sodasql/dialects/hive_dialect.py

```diff
@@ -0,0 +1,123 @@
+from pyhive import hive
+from pyhive.exc import Error
+from thrift.transport.TTransport import TTransportException
+
+from sodasql.exceptions.exceptions import WarehouseConnectionError, WarehouseAuthenticationError
+from sodasql.scan.dialect import Dialect, HIVE, KEY_WAREHOUSE_TYPE
+from sodasql.scan.parser import Parser
+import json
+
+class HiveDialect(Dialect):
+    data_type_decimal = "DECIMAL"
+
+    def __init__(self, parser: Parser):
+        super().__init__(HIVE)
+        if parser:
+            self.host = parser.get_str_required('host')
+            self.port = parser.get_int_optional('port', '10000')
+            self.username = parser.get_str_required_env('username')
+            self.password = parser.get_str_required_env('password')
+            self.database = parser.get_str_optional('database', 'default')
+            self.configuration = parser.get_dict_optional('configuration')
+
+    def default_connection_properties(self, params: dict):
+        return {
+            KEY_WAREHOUSE_TYPE: HIVE,
+            'host': 'localhost',
+            'port': 10000,
+            'username': 'env_var(HIVE_USERNAME)',
+            'password': 'env_var(HIVE_PASSWORD)',
+            'database': params.get('database', 'your_database')
+        }
+
+    def default_env_vars(self, params: dict):
+        return {
+            'HIVE_USERNAME': params.get('username', 'hive_username_goes_here'),
+            'HIVE_PASSWORD': params.get('password', 'hive_password_goes_here')
+        }
+
+    def sql_tables_metadata_query(self, limit: str = 10, filter: str = None):
+        return (f"use {self.database.lower()};\n"
+                f"show tables;")
+
+    def create_connection(self, *args, **kwargs):
+        self.configuration['hive.ddl.output.format']='json'
+        try:
+            conn = hive.connect(
+                username=self.username,
+                password=self.password,
+                host=self.host,
+                port=self.port,
+                database=self.database,
+                configuration={ key: str(value) for key, value in self.configuration.items() }, #https://github.com/jaegertracing/jaeger-client-python/issues/151
+                auth=None)
+            return conn
+        except Exception as e:
+            self.try_to_raise_soda_sql_exception(e)
+
+    def sql_columns_metadata(self, table_name:str):
+        # getting columns info from hive which version <3.x needs to be parsed
+        column_tuples = []
+        cur = self.create_connection().cursor()
```

Comment:
Can you confirm if this connection will be closed after the query? If not, can you call close at the end of this function to make sure that it is cleaned up?

Reply:
done